src/Mvc/Mvc.Core/src/ModelBinding/Validation/ValidationVisitor.cs

```diff
@@ -223,13 +223,24 @@ protected virtual bool Visit(ModelMetadata metadata, string key, object model)
                 return true;
             }
 
+            bool result;
+            try
+            {
+                result = VisitImplementation(ref metadata, ref key, model);
+            }
+            finally
+            {
+                _currentPath.Pop(model);
```

Comment:
It would be probably be helpful to have a comment here about what we expect to throw.

Reply:
Done. Is that the right place for the comment?

Reply:
Looks good.